### PR TITLE
[Minor][MLLIB] Refactor toString method in MLLIB

### PR DESCRIPTION
**Pull request by aihex**
1. predict(predict.toString) has already output prefix “predict” thus it’s duplicated to print ", predict = " again
2. there are some extra spaces

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/tree/model/Node.scala [hunk 1]
```diff
@@ -51,8 +51,8 @@ class Node (
     var stats: Option[InformationGainStats]) extends Serializable with Logging {
 
   override def toString: String = {
-    "id = " + id + ", isLeaf = " + isLeaf + ", predict = " + predict + ", " +
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/tree/model/Node.scala]: These can use string interpolation. I take your point though it breaks the symmetry a bit and make this `toString` rely on details of the subclass. How about making the `Predict.toString` return something more compact like `s"$predict ($prob)"`?

**Comment by AiHe** [comment]
Modify Predict.scala and decouple the dependency of Node.scala on Predict.scala.

**Comment by srowen** [comment]
I'm OK with this, though while you're at it, why not make these use string interpolation? `s"foo = $foo, ..."`

**Comment by AiHe** [comment]
If you're referring to the modification on Predict.scala, I just follow the style of previous code.

**Comment by srowen** [comment]
Yes, but that's a hold over from Scala 2.9 and before. I think that while we're modifying these strings they can use the 'modern' style used in new code in the project.

**Comment by AiHe** [comment]
Okay, sounds better to use a modern style.

**Comment by srowen** [comment]
... this still is not using string interpolation everywhere. It's a small thing, but if we're bothering to update these `toString` methods, go all the way.

File: mllib/src/main/scala/org/apache/spark/mllib/tree/model/Predict.scala [hunk 2]
```diff
@@ -29,9 +29,7 @@ class Predict(
     val predict: Double,
     val prob: Double = 0.0) extends Serializable {
 
-  override def toString: String = {
-    "predict = %f, prob = %f".format(predict, prob)
-  }
+  override def toString: String = s"$predict(prob = $prob)"
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/tree/model/Predict.scala]: I'd put a space before the parenthesis: ```$predict (prob```

**Comment by AiHe** [comment]
Get your point and change all toString methods in MLLIB. 
There are a large number of uses of 'old' way in the statement like logInfo, which makes it hard to change all them in a moment.

**Comment by srowen** [comment]
Looks OK, I think Joseph just had one more tiny comment on one `toString` method, to add a space.

**Comment by AiHe** [comment]
Good to go. 
@jkbradley Thank you.

**Comment by srowen** [comment]
ok to test

**Comment by jkbradley** [comment]
LGTM pending tests

**Comment by srowen** [comment]
@AiHe  this fails style checks. I think the lines are too long. Can you fix those up and try `./dev/lint-scala` to check it out locally?

**Comment by AiHe** [comment]
@srowen Sure. Pass the style checking and will do that for PRs in the feature.

**Pull request closed without merging**